src/app/screens/SignTransaction/SignTransaction.tsx

```diff
@@ -36,9 +36,22 @@ export const SignTransaction = () => {
   }
 
   if (currentAccount) {
+    if (currentAccount.type === WalletType.MNEMONIC) {
```

Comment:
Why is this necessary? This was much cleaner without this.

Reply:
To properly infer the type of `currentAccount`. Otherwise, the type would be inferred as `never`. Alternatively I can do a type cast if you think that's better?

Reply:
I would prefer that over this yeah.